src/server/JasmineGraphInstanceService.cpp

```diff
@@ -3584,3 +3664,308 @@ map<long, map<long, unordered_set<long>>> calculateEgoNet(string graphID, string
+        long vertex = hMapAuthorityFlowWorldToLocalIterator.first;
+        float worldToLocalRank = hMapAuthorityFlowWorldToLocalIterator.second;
+
+        auto rankMapItr = rankMap.find(vertex);
+        if (rankMapItr != rankMap.end()) {
+            double existingRank = rankMapItr->second;
+            double finalRank = existingRank + worldToLocalRank;
+            rankMapItr->second = finalRank;
+        } else {
+            rankMap.insert(std::make_pair(vertex, worldToLocalRank));
+        }
+    }
+
+    int count = 0;
+    while (count < ITERATION_COUNT) {
+
+        for (localGraphMapIterator = localGraphMap.begin();
+             localGraphMapIterator != localGraphMap.end(); ++localGraphMapIterator) {
+
+            long startVid = localGraphMapIterator->first;
+            unordered_set<long> endVidSet = localGraphMapIterator->second;
+            double existingParentRank = 1;
+
+            auto rankMapItr = rankMap.find(startVid);
+            if (rankMapItr != rankMap.end()) {
+                existingParentRank = rankMapItr->second;
+            } else {
+                rankMap.insert(std::make_pair(startVid, existingParentRank));
+            }
+
+            long degree = endVidSet.size();
+            double distributedRank = alpha * (existingParentRank / degree) + mu;
+
+            for (long itr: endVidSet) {
+                auto rankMapItr = rankMap.find(itr);
+
+                double existingChildRank = 0;
+                double finalRank = 0;
+                if (rankMapItr != rankMap.end()) {
+                    existingChildRank = rankMapItr->second;
+                    finalRank = existingChildRank + distributedRank;
+
+                    rankMapItr->second = finalRank;
+                } else {
+                    finalRank = existingChildRank + distributedRank;
+                    rankMap.insert(std::make_pair(itr, finalRank));
+
+                }
+            }
+        }
+
+        count++;
+    }
+
+    map<double, long> rankMapResults;
+
+    if (top_k_page_rank_value == -1) {
+        instance_logger.log("Page rank is not implemented", "info");
+    } else {
+        std::string resultTree = "";
+
+        for (map<long, double>::iterator rankMapItr = rankMap.begin(); rankMapItr != rankMap.end(); ++rankMapItr) {
+            rankMapResults.insert(std::make_pair(rankMapItr->second, rankMapItr->first));
+            resultTree.append(to_string(rankMapItr->first) + ":" + to_string(rankMapItr->second) + ",");
+        }
+
+        int count = 0;
+        for (map<double, long>::iterator rankMapItr = rankMapResults.end();
+             rankMapItr != rankMapResults.begin(); --rankMapItr) {
+
+            if (count > top_k_page_rank_value) {
+                break;
+            }
+            instance_logger.log(to_string(rankMapItr->first) + ":" + to_string(rankMapItr->second) + ",", "info");
+            count++;
+        }
+
```

Comment:
There seems some unnecessary new lines between the braces.

Reply:
fixed

Reply:
Ok